File: PayTR Direkt API/PayTR Direkt API/Direkt API - Türkçe/1. ADIM/1. ADIM .PHP Örnek/direct_api_ornek.php
<!doctype html>
<html>
	<head>
		<meta charset="UTF-8">
	</head>
	
	<?php

		$merchant_id = 'MAGAZA_NO';
		$merchant_key = 'XXXXXXXXXXX';
		$merchant_salt = 'YYYYYYYYYYY';

		$merchant_ok_url="http://site-ismi/basarili";
		$merchant_fail_url="http://site-ismi/basarisiz";

        $user_basket = htmlentities(json_encode(array(
            array("Altis Renkli Deniz Yatağı - Mavi", "18.00", 1),
            array("Pharmasol Güneş Kremi 50+ Yetişkin & Bepanthol Cilt Bakım Kremi", "33,25", 2),
            array("Bestway Çocuklar İçin Plaj Seti Beach Set ÇANTADA DENİZ TOPU-BOT-KOLLUK", "45,42", 1)
        )));

		srand(time(null));
		$merchant_oid = rand();

		$test_mode="0";

		//3d'siz işlem
		$non_3d="0";

		//Ödeme süreci dil seçeneği tr veya en
        $client_lang = "tr";

		//non3d işlemde, başarısız işlemi test etmek için 1 gönderilir (test_mode ve non_3d değerleri 1 ise dikkate alınır!)
		$non3d_test_failed="0";

        if( isset( $_SERVER["HTTP_CLIENT_IP"] ) ) {
            $ip = $_SERVER["HTTP_CLIENT_IP"];
        } elseif( isset( $_SERVER["HTTP_X_FORWARDED_FOR"] ) ) {
            $ip = $_SERVER["HTTP_X_FORWARDED_FOR"];
        } else {
            $ip = $_SERVER["REMOTE_ADDR"];
        }

        $user_ip = $ip;

		$email = "testnon3d@paytr.com";

		// 100.99 TL ödeme
		$payment_amount = "100.99";
		$currency="TL";
		//
        $payment_type = "card";


//		$card_type = "bonus";       // Alabileceği değerler; advantage, axess, combo, bonus, cardfinans, maximum, paraf, world, saglamkart
//		$installment_count = "5";

        $post_url = "https://www.paytr.com/odeme";

		$hash_str = $merchant_id . $user_ip . $merchant_oid . $email . $payment_amount . $payment_type . $installment_count. $currency. $test_mode. $non_3d;
		$token = base64_encode(hash_hmac('sha256',$hash_str.$merchant_salt,$merchant_key,true));
	?>
	
	<body>
		<form action="<?php echo $post_url;?>" method="post">
		  Kart Sahibi Adı: <input type="text" name="cc_owner" value="TEST KARTI"><br>
		  Kart Numarası: <input type="text" name="card_number" value="9792030394440796"><br>
		  Kart Son Kullanma Ay: <input type="text" name="expiry_month" value="12" ><br>
		  Kart Son Kullanma Yıl: <input type="text" name="expiry_year" value="99"><br>
		  Kart Güvenlik Kodu: <input type="text" name="cvv" value="000"><br>
		  <input type="hidden" name="merchant_id" value="<?php echo $merchant_id;?>">
		  <input type="hidden" name="user_ip" value="<?php echo $user_ip;?>">
		  <input type="hidden" name="merchant_oid" value="<?php echo $merchant_oid;?>">
		  <input type="hidden" name="email" value="<?php echo $email;?>">
		  <input type="hidden" name="payment_type" value="<?php echo $payment_type;?>">
		  <input type="hidden" name="payment_amount" value="<?php echo $payment_amount;?>">
		  <input type="hidden" name="currency" value="<?php echo $currency;?>">
		  <input type="hidden" name="test_mode" value="<?php echo $test_mode;?>">
		  <input type="hidden" name="non_3d" value="<?php echo $non_3d;?>">
		  <input type="hidden" name="merchant_ok_url" value="<?php echo $merchant_ok_url;?>">
		  <input type="hidden" name="merchant_fail_url" value="<?php echo $merchant_fail_url;?>">
		  <input type="hidden" name="user_name" value="Paytr Test">
		  <input type="hidden" name="user_address" value="test test test">
		  <input type="hidden" name="user_phone" value="05555555555">
		  <input type="hidden" name="user_basket" value="<?php echo $user_basket; ?>">
		  <input type="hidden" name="debug_on" value="1">
		  <input type="hidden" name="client_lang" value="<?php echo $client_lang; ?>">
		  <input type="hidden" name="paytr_token" value="<?php echo $token; ?>">
		  <input type="hidden" name="non3d_test_failed" value="<?php echo $non3d_test_failed; ?>">
		  <input type="hidden" name="installment_count" value="<?php echo $installment_count; ?>">
		  <input type="hidden" name="card_type" value="<?php echo $card_type; ?>">
		  <input type="submit" value="Submit">
		</form>
	</body>
</html>

File: PayTR Direkt API/PayTR Direkt API/Direkt API - Türkçe/1. ADIM/1. ADIM PYTHON Örnek/direct_api_ornek.py
# Python 3.6+
# Django Web Framework referans alınarak hazırlanmıştır
# Tek başına bir bütün değildir, home.html ile birlikte çalışmaktadır.
# card_type, installment_count gibi kullanıcıya bağlı bilgiler alındıktan sonra paytr_token oluşturulması gerekmektedir.

import base64
import hashlib
import hmac
import html
import json
import random

from django.shortcuts import render, HttpResponse
from django.views.decorators.csrf import csrf_exempt


def home(request):
    merchant_id = 'MAGAZA_NO'
    merchant_key = b'XXXXXXXXXXX'
    merchant_salt = b'YYYYYYYYYYY'

    merchant_ok_url = 'http://site-ismi/basarili'
    merchant_fail_url = 'http://site-ismi/basarisiz'

    user_basket = html.unescape(json.dumps([['Altis Renkli Deniz Yatağı - Mavi', '18.00', 1],
                                            ['Pharmaso Güneş Kremi 50+ Yetişkin & Bepanthol Cilt Bakım Kremi', '33,25',
                                             2],
                                            ['Bestway Çocuklar İçin Plaj Seti Beach Set ÇANTADA DENİZ TOPU-BOT-KOLLUK',
                                             '45,42', 1]]))

    merchant_oid = 'OS' + random.randint(1, 9999999).__str__()
    test_mode = '0'
    debug_on = '1'

    # 3d'siz işlem
    non_3d = '0'

    # Ödeme süreci dil seçeneği tr veya en
    client_lang = 'tr'

    # non3d işlemde, başarısız işlemi test etmek için 1 gönderilir (test_mode ve non_3d değerleri 1 ise dikkate alınır!)
    non3d_test_failed = '0'
    user_ip = ''
    email = 'testnon3d@paytr.com'

    # 100.99 TL ödeme
    payment_amount = "100.99"
    currency = 'TL'
    payment_type = 'card'

    user_name = 'Paytr Test'
    user_address = 'test test test'
    user_phone = '05555555555'

    # Alabileceği değerler; advantage, axess, combo, bonus, cardfinans, maximum, paraf, world
    card_type = 'bonus'
    installment_count = '5'

    hash_str = merchant_id + user_ip + merchant_oid + email + payment_amount + payment_type + installment_count + currency + test_mode + non_3d
    paytr_token = base64.b64encode(hmac.new(merchant_key, hash_str.encode() + merchant_salt, hashlib.sha256).digest())

    context = {
        'merchant_id': merchant_id,
        'user_ip': user_ip,
        'merchant_oid': merchant_oid,
        'email': email,
        'payment_type': payment_type,
        'payment_amount': payment_amount,
        'currency': currency,
        'test_mode': test_mode,
        'non_3d': non_3d,
        'merchant_ok_url': merchant_ok_url,
        'merchant_fail_url': merchant_fail_url,
        'user_name': user_name,
        'user_address': user_address,
        'user_phone': user_phone,
        'user_basket': user_basket,
        'debug_on': debug_on,
        'client_lang': client_lang,
        'paytr_token': paytr_token.decode(),
        'non3d_test_failed': non3d_test_failed,
        'installment_count': installment_count,
        'card_type': card_type
    }

    return render(request, 'home.html', context)